### PR TITLE
Switching to poetry and removing support for python 2

### PR DESCRIPTION
**Pull request by ilaner**
## Description
This PR removes the old dev-requiements-py*.txt, switches the content dependencies to poetry and removing support for local python2 to be installed.

### CODE REVIEW

File: .gitlab/ci/global.yml [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
       files:
-        - "dev-requirements-py3.txt"
+        - "poetry.lock"
         - "package-lock.json"
-      prefix: $CI_COMMIT_REF_SLUG
+      prefix: dev-content
     paths:
       - $PIP_CACHE_DIR
-      - venv/
+      - .venv
```
Comment (tkatzir) [inline .gitlab/ci/global.yml:10]: `.venv` or `.venv/` ?

**Review by tkatzir — APPROVED** [review]

**Comment by ilaner** [comment]
Builds:
Upload flow: https://code.pan.run/xsoar/content/-/pipelines/3054760
Content nightly: https://code.pan.run/xsoar/content/-/pipelines/3054766
SDK nightly: https://code.pan.run/xsoar/content/-/pipelines/3054768

File: .circleci/config.yml [hunk 2]
```diff
@@ -94,7 +94,7 @@ references:
 
   container_config: &container_config
     docker:
-      - image: devdemisto/gitlab-content-ci:1.0.0.30459  # disable-secrets-detection
```
Comment (jochman) [inline .circleci/config.yml]: that's a good change

**Comment by ilaner** [comment]
another shot:
content nightly - https://code.pan.run/xsoar/content/-/pipelines/3075902
test upload - https://code.pan.run/xsoar/content/-/pipelines/3075903
sdk nightly - https://code.pan.run/xsoar/content/-/pipelines/3075904

File: .circleci/config.yml [hunk 3]
```diff
@@ -94,7 +94,7 @@ references:
 
   container_config: &container_config
     docker:
-      - image: devdemisto/content-build:3.0.0.30473  # disable-secrets-detection
+      - image: devdemisto/content-build:3.0.0.30555  # disable-secrets-detection
```
Comment (jochman) [inline .circleci/config.yml:97]: 555
:hamsa:

File: .gitlab/ci/global.yml [hunk 4]
```diff
@@ -158,7 +158,7 @@
 .default-job-settings:
   interruptible: true
   extends:
-    - .default-cache
+    # - .default-cache
```
Comment (dorschw) [inline .gitlab/ci/global.yml:161]: intentional?

Comment (ilaner) [reply .gitlab/ci/global.yml]: was intentional for testing, now we can restore the cache

**Comment by tkatzir** [comment]
Switching to poetry and removing support for python 2

**Pull request merged**